scene/resources/skeleton_modification_2d.h

```diff
@@ -0,0 +1,620 @@
+/*************************************************************************/
+/*  skeleton_modification_2d.h                                           */
```

Comment:
This file is really too big. Would be much better separate each `modifier` (algorithm) in its own `.h`/`.cpp`: This can improve the code organization, so each file has its own topic, and eventual comments that explain an algorithm or a piece of code are well contained.

Reply:
I totally agree and was going to use separate files initially, but when I was looking through the Godot source code when I started implementing this, I found there was a pattern where large files were used to contain everything that could be grouped together. Examples of this are the CanvasItem gizmo file, the physics joints, and the Skeleton2D and Bone2D file, among others. Granted, this was last I checked and that was before GSoC 2020, so things may have changed since then.

I think this would make a great future PR though, but I'm not keen on making the change currently because of the effect it would have on the Git history.

Reply:
It would make sense to split the classes in different files for readability. What problem would it cause on the git history? The file is new and the commits are going to be squashed before merging anyway.